src/Arcus.Security.Providers.AzureKeyVault/Configuration/SecretStoreBuilderExtensions.cs

```diff
@@ -0,0 +1,100 @@
+using Arcus.Security.Providers.AzureKeyVault;
+using Arcus.Security.Providers.AzureKeyVault.Authentication;
+using Arcus.Security.Providers.AzureKeyVault.Configuration;
+using GuardNet;
+
+// ReSharper disable once CheckNamespace
+namespace Microsoft.Extensions.Hosting
+{
+    /// <summary>
+    /// Extensions on the <see cref="SecretStoreBuilder"/> to provide easy addition the Azure Key Vault secrets in the secret store.
+    /// </summary>
+    public static class SecretStoreBuilderExtensions
+    {
+        /// <summary>
+        /// Adds a secret source to the secret store of the application that gets its secrets from Azure Key Vault.
+        /// </summary>
+        /// <param name="builder">The builder to create the secret store.</param>
+        /// <param name="rawVaultUri">The Uri of the Azure Key Vault you want to connect to.</param>
+        /// <param name="clientId">The identifier of the application requesting the authentication token.</param>
+        /// <param name="certificate">The certificate that is used as credential.</param>
+        public static SecretStoreBuilder AddAzureKeyVaultWithCertificate(
+            this SecretStoreBuilder builder,
+            string rawVaultUri,
+            string clientId,
+            X509Certificate2 certificate)
+        {
+            Guard.NotNull(rawVaultUri, nameof(rawVaultUri));
+            Guard.NotNull(clientId, nameof(clientId));
+            Guard.NotNull(certificate, nameof(certificate));
+
+            return AddAzureKeyVault(
+                builder,
+                new CertificateBasedAuthentication(clientId, certificate),
+                new KeyVaultConfiguration(rawVaultUri));
+        }
+
+        /// <summary>
+        /// Adds a secret source to the secret store of the application that gets its secrets from Azure Key Vault.
+        /// </summary>
+        /// <param name="builder">The builder to create the secret store.</param>
+        /// <param name="rawVaultUri">The Uri of the Azure Key Vault you want to connect to.</param>
+        /// <param name="connectionString">The connection string to use to authenticate, if applicable.</param>
+        /// <param name="azureADInstance">The azure AD instance to use to authenticate, if applicable.</param>
+        public static SecretStoreBuilder AddAzureKeyVaultWithManagedServiceIdentity(
+            this SecretStoreBuilder builder,
+            string rawVaultUri,
+            string connectionString = null,
+            string azureADInstance = null)
+        {
+            Guard.NotNull(rawVaultUri, nameof(rawVaultUri));
+
+            return AddAzureKeyVault(
+                builder,
+                new ManagedServiceIdentityAuthentication(connectionString, azureADInstance),
+                new KeyVaultConfiguration(rawVaultUri));
+        }
+
+        /// <summary>
+        /// Adds a secret source to the secret store of the application that gets its secrets from Azure Key Vault.
```

Comment:
Was more suggesting to use something similar to this:
```suggestion
        /// Adds Azure Key Vault as a secret source which uses client secret authentication.
```